components/camera/camera.go

```diff
@@ -228,6 +228,17 @@ func SourceFromCamera(cam Camera) (gostream.VideoSource, error) {
 	if asSrc, ok := cam.(*videoSource); ok {
 		return asSrc.videoSource, nil
 	}
+
+	if asWaitGroup, ok := cam.(*CameraWaitGroup); ok {
+		asAct := utils.UnwrapProxy(asWaitGroup.Cam).(Camera)
```

Comment:
We don't know if the underlying camera (`asWaitGroup.Cam` here) will always be a `reconfigurableCamera` so we don't need to call `utils.UnwrapProxy` here. If it _is_, then a recursive call to `SourceFromCamera(asWaitGroup.Cam)` will unwrap it making this redundant. If it is _not_ then this call will do nothing and is essentially a no-op.